packages/web-components/src/button/button.ts

```diff
@@ -0,0 +1,66 @@
+import { attr } from '@microsoft/fast-element';
+import { FASTButton } from '@microsoft/fast-foundation';
+import { ButtonAppearance, ButtonShape, ButtonSize } from './button.options.js';
+
+/**
+ * The base class used for constructing a fluent-button custom element
+ * @public
+ */
+export class Button extends FASTButton {
+  /**
+   * The appearance the button should have.
+   *
+   * @public
+   * @remarks
+   * HTML Attribute: appearance
+   */
+  @attr
+  public appearance?: ButtonAppearance | undefined;
+
+  /**
+   * The shape the button should have.
+   *
+   * @public
+   * @remarks
+   * HTML Attribute: shape
+   */
+  @attr
+  public shape?: ButtonShape | undefined;
+
+  /**
+   * The size the button should have.
+   *
+   * @public
+   * @remarks
+   * HTML Attribute: size
+   */
+  @attr
+  public size?: ButtonSize;
+
+  /**
+   * The button has an icon only, no text content
+   *
+   * @public
+   * @remarks
+   * HTML Attribute: icon-only
+   */
+  @attr({ attribute: 'icon-only', mode: 'boolean' })
+  public iconOnly?: boolean = false;
+
+  /**
+   * The button is disabled but focusable
+   *
+   * @public
+   * @remarks
+   * HTML Attribute: disabled-focusable
+   */
+  @attr({ attribute: 'disabled-focusable', mode: 'boolean' })
+  public disabledFocusable?: boolean = false;
+  protected disabledFocusableChanged(prev: boolean, next: boolean): void {
+    if (this.disabledFocusable) {
+      ((this as unknown) as HTMLElement).setAttribute('aria-disabled', 'true');
```

Comment:
This is obviously unique to our implementation. Because we learned that browsers wont allow focusing even when setting `tabIndex="0"` if  `disabled` is present. Except Safari

Reply:
The key consideration here is that we aren't setting this to disabled, which is why this will maintain focus - this is a good reminder though that we should likely suppress events in the case of `disabled-focusable`...